Dockerfile

```
@@ -0,0 +1,32 @@
+# Perform common operations, dependency installation etc...
+FROM python:3.12.1 AS base
+# Install poetry
+RUN curl -sSL https://install.python-poetry.org | python3 -
+# Add Environment variable
+ENV PATH=$PATH:/root/.local/bin/
+# Set work directory
+WORKDIR /
+# Copy across pyproject.toml
+COPY pyproject.toml /
```

Here we also want to copy in the `poetry.lock` file, otherwise we risk ending up with a slightly different dependency tree each time, which could lead to subtle/inconsistent bugs
```suggestion
COPY poetry.lock pyproject.toml /
```